Sources/SDK/Service/HttpClient.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+typealias HttpHeaders = [String: String]
+
+internal protocol HttpClient: AutoMockable {
+    func request(
+        _ endpoint: HttpEndpoint,
+        headers: HttpHeaders?,
+        body: Data?,
+        onComplete: @escaping (Result<Data, HttpRequestError>) -> Void
+    )
+}
+
+internal class CIOHttpClient: HttpClient {
+    private let session: URLSession
+    private let region: Region
+    private var httpRequestRunner: HttpRequestRunner
+
+    /// for testing
+    init(httpRequestRunner: HttpRequestRunner, region: Region) {
+        self.httpRequestRunner = httpRequestRunner
+        self.session = Self.getSession(siteId: "fake-site-id", apiKey: "fake-api-key")
+        self.region = region
+    }
+
+    init(config: SdkConfig) {
+        self.session = Self.getSession(siteId: config.siteId, apiKey: config.apiKey)
+        self.region = config.region
```

Comment:
The idea behind having the Region stored is:

`HttpEndpoint` knows everything about how to construct a URL string. Each endpoint knows the HTTP method, path, and what base URL string the endpoint belongs to (the tracking API vs App API for example). 

That's the reasoning behind needing to store the region because when `request(HttpEndpoint)` gets called, we ask the `HttpEndpoint` parameter for the full URL (which requires knowing the Region to get the base tracking URL string, base App API URL string, etc)

This logic is assuming that the mobile SDK could use more then just the tracking API in the future. If we feel confident the mobile SDK will only ever call 1 of our APIs (the tracking API) that is how I believe we can easily remove storing the Region instance.

Reply:
:cool: seems fine. In our other clients we handle that by setting both urls at config time, for both track and ext_api. Don't think it matters too much, if it becomes cumbersome in future we can refactor. :+1:

Reply:
Didn't think about that idea - setting the URLs when region gets set during configuration time. 

I believe this goes along with the other comments about configuring the URL to be something other then production. I believe we can fix both issues with that change.